File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsJava/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicsjava/ui/UserViewModel.java
package pl.udu.uwr.pum.offlinecachingbasicsjava.ui;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;
import pl.udu.uwr.pum.offlinecachingbasicsjava.data.User;
import pl.udu.uwr.pum.offlinecachingbasicsjava.data.repository.UserRepository;
import pl.udu.uwr.pum.offlinecachingbasicsjava.util.Resource;

@HiltViewModel
public class UserViewModel extends ViewModel {
    private LiveData<Resource<List<User>>> users;
    private final UserRepository repository;

    @Inject
    public UserViewModel(UserRepository repository) {
        this.repository = repository;

    }

    public LiveData<Resource<List<User>>> getUsers() {
        return repository.getUsers();
    }
}

File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsKotlin/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicskotlin/ui/UserViewModel.kt
package pl.udu.uwr.pum.offlinecachingbasicskotlin.ui

import androidx.lifecycle.ViewModel
import androidx.lifecycle.asLiveData
import dagger.hilt.android.lifecycle.HiltViewModel
import pl.udu.uwr.pum.offlinecachingbasicskotlin.data.repository.UserRepository
import javax.inject.Inject

@HiltViewModel
class UserViewModel @Inject constructor(
    repository: UserRepository
) : ViewModel() {
    val users = repository.getUsers().asLiveData()
}